kratos/utilities/shifted_boundary_meshless_interface_utility.cpp

```diff
@@ -0,0 +1,1124 @@
+        block_for_each(mpModelPart->Elements(), [](Element& rElement){
+            rElement.Set(ACTIVE, false); // Elements in the positive distance region (the ones to be assembled)
+            rElement.Set(BOUNDARY, false); // Intersected elements
+            rElement.Set(INTERFACE, false); // Positive distance elements owning the surrogate boundary nodes
+        });
+
+        // Find active regions
+        for (auto& rElement : mpModelPart->Elements()) {
+            auto& r_geom = rElement.GetGeometry();
+            if (IsSplit(r_geom)) {
+                // Mark the intersected elements as BOUNDARY
+                // The intersected elements are also flagged as non ACTIVE to avoid assembling them
+                // Note that the split elements BC is applied by means of the extension operators
+                rElement.Set(ACTIVE, false);
+                rElement.Set(BOUNDARY, true);
+                // Mark the positive intersected nodes as BOUNDARY
+                for (auto& rNode : r_geom) {
+                    if (!(rNode.FastGetSolutionStepValue(DISTANCE) < 0.0)) {
+                        rNode.Set(BOUNDARY, true);
+                    }
+                }
+            } else if (IsNegative(r_geom)) {
+                // Mark the negative element as non ACTIVE to avoid assembling it
+                rElement.Set(ACTIVE, false);
+            } else {
+                // Mark the positive element as ACTIVE to assemble it
+                rElement.Set(ACTIVE, true);
+                // Mark the active element nodes as ACTIVE
+                for (auto& rNode : r_geom) {
+                    rNode.Set(ACTIVE, true);
+                }
+            }
+        }
+
+        // Find the surrogate boundary elements
+        // Note that we rely on the fact that the neighbours are sorted according to the faces
+        for (auto& rElement : mpModelPart->Elements()) {
+            if (rElement.Is(BOUNDARY)) {
+                const auto& r_geom = rElement.GetGeometry();
+                const std::size_t n_faces = r_geom.FacesNumber();
+                auto& r_neigh_elems = rElement.GetValue(NEIGHBOUR_ELEMENTS);
+                for (std::size_t i_face = 0; i_face < n_faces; ++i_face) {
+                    // The neighbour corresponding to the current face is ACTIVE means that the current face is surrogate boundary
+                    // Flag the current neighbour owning the surrogate face as INTERFACE
+                    // The nodes will be flagged if required (MLS basis) when creating the cloud
+                    auto p_neigh_elem = r_neigh_elems(i_face).get();
+                    if (p_neigh_elem != nullptr) {
+                        if (p_neigh_elem->Is(ACTIVE)) {
+                            p_neigh_elem->Set(INTERFACE, true);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    bool ShiftedBoundaryMeshlessInterfaceUtility::IsSplit(const GeometryType& rGeometry)

```

Comment:
```suggestion
    bool ShiftedBoundaryMeshlessInterfaceUtility::IsSplit(const GeometryType& rGeometry) const

```

or better, make it a free function at the top of the file

Reply:
Nice suggestion. I put them as free functions in an anonymous namespace. Indeed, my idea is to migrate these to an `DistanceFunctionUtilities` or something of this sort (they are repeated in many places...).